packages/components/src/hooks/useInteractionScriptLoader.ts

```diff
@@ -0,0 +1,33 @@
+import { useRef, useState } from "react"
+import { useEventListener, useScript, useTimeout } from "usehooks-ts"
+
+interface UseInteractionScriptLoaderOptions {
+  src: string
+  timeout?: number
+}
+
+// TBT flags any script blocking the main thread for >50ms.
+// Large third-party widgets (Vica ~800kb, AskGov ~250kb) exceed this,
+// and Lighthouse still counts them even with async
+// Thus we use a custom loader hook to delay loading until after Wogaa has stopped measuring TBT.
+export const useInteractionScriptLoader = ({
+  src,
+  timeout = 3000, // 3 seconds from manual testing
+}: UseInteractionScriptLoaderOptions) => {
+  const documentRef = useRef<Document | null>(
+    typeof document !== "undefined" ? document : null,
+  )
+  const [shouldLoad, setShouldLoad] = useState(false)
+
+  const triggerLoad = () => setShouldLoad(true)
+
+  // Load script on user interactions (scroll, click, touchstart)
+  useEventListener("scroll", triggerLoad, documentRef, { passive: true })
```

Comment:
should we also do a cleanup of these listeners on unmount or is it handled by the lib?